x/concentrated-liquidity/incentives.go

```diff
@@ -340,3 +340,76 @@ func (k Keeper) GetUptimeGrowthOutsideRange(ctx sdk.Context, poolId uint64, lowe
 
 	return osmoutils.SubDecCoinArrays(globalUptimeValues, uptimeGrowthInside)
 }
+
+// initOrUpdatePositionUptime either adds or updates records for all uptime accumulators `position` qualifies for
+func (k Keeper) initOrUpdatePositionUptime(ctx sdk.Context, poolId uint64, position *model.Position, owner sdk.AccAddress, lowerTick, upperTick int64, liquidityDelta sdk.Dec, frozenUntil time.Time) error {
```

Comment:
I know this was discussed before but can we separate init with update? Maybe in a post-completion refactor?

Currently, we are bundling two separate behaviors together, increasing the complexity and a likelihood of missing something small.

For ref, with fees, we init the position separately here: https://github.com/osmosis-labs/osmosis/blob/03f2a989a5b603ad9b34b90294bcbb7c7dae86cc/x/concentrated-liquidity/lp.go#L72-L74

and then doing the update here assuming that it is already initialized: https://github.com/osmosis-labs/osmosis/blob/03f2a989a5b603ad9b34b90294bcbb7c7dae86cc/x/concentrated-liquidity/lp.go#L225-L227

Reply:
I believe that after #4468 is merged and positions are keyed by `joinTime` and `freezeDuration` (instead of just `frozenUntil`, we will be able to safely remove the ability to add to a frozen position, as the only case where it would be relevant becomes when someone tries to deposit to the same `freezeDuration` multiple times in the same block (see #4482).

Since this is not functionality we necessarily need to support (and actually disallowing it would serve as a great spam resistance vector), I think it will be appropriate to simplify this function to only handle initialization.